src/main/java/org/jenkinsci/plugins/github_branch_source/Connector.java

```diff
@@ -108,4 +111,25 @@ private static CredentialsMatcher githubScanCredentialsMatcher() {
 
     private Connector() {}
 
+    /**
+     * Fail immediately and throw a customized exception.
+     */
+    public static final RateLimitHandler CUSTOMIZED = new RateLimitHandler() {
```

Comment:
@abayer yes, it what exactly i fixed in github-api. Either it Thread.sleeps() with wait either fails. In latest github-api version something was changed in fail logic (order of checks).  I.e. if you provided wrong creds, you should be able to fail and pick new connector. (Most of issues was verified in my pr plugin and resolved in api).